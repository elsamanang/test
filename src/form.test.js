import renderer from 'react-test-renderer';
import From from './component/form';

it('Charge the form', () => {

    const fakeForm =[
        {
          "name": "KnownErrorTypeId",
          "displayName": "Knowledge Item Type",
          "type": "select",
          "x-options": [
            { "value": 1, "text": "FAQ" },
            { "value": 2, "text": "Advice" },
            { "value": 3, "text": "Known Error" }
          ]
        },
        {
          "name": "Status",
          "displayName": "Lifecycle Status",
          "type": "select",
          "x-options": [
            { "value": 1, "text": "Published" },
            { "value": 2, "text": "Planned" },
            { "value": 3, "text": "Retired" }
          ]
        },
        {
          "name": "IsPrivate",
          "displayName": "Is Private?",
          "type": "checkbox"
        },
        {
          "name": "Summary",
          "displayName": "Summary",
          "type": "text"
        }
    ]

    const fakeSelcted ={
        KnownErrorTypeId: 1,
        Status: 1,
        IsPrivate: true,
        Summary: "element summary"
    }
    
    const component = renderer.create(
        <From formulaire={fakeForm} selected={fakeSelcted} />,
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

})