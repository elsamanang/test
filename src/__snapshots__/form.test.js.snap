// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Charge the form 1`] = `
<div
  className="content"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <form>
        <div
          className="form-group"
        >
          <label>
            Knowledge Item Type
          </label>
          <select
            className="form-control m-1"
            id="exampleFormControlSelect1"
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              FAQ
            </option>
            <option
              value={2}
            >
              Advice
            </option>
            <option
              value={3}
            >
              Known Error
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label>
            Lifecycle Status
          </label>
          <select
            className="form-control m-1"
            id="exampleFormControlSelect1"
            onChange={[Function]}
            value={1}
          >
            <option
              value={1}
            >
              Published
            </option>
            <option
              value={2}
            >
              Planned
            </option>
            <option
              value={3}
            >
              Retired
            </option>
          </select>
        </div>
        <div
          className="form-group"
        >
          <label>
            Is Private?
          </label>
          <input
            checked={true}
            className="form-check-input m-1"
            id="exampleInputEmail1"
            name="name"
            onChange={[Function]}
            type="checkbox"
            value={true}
          />
        </div>
        <div
          className="form-group"
        >
          <label>
            Summary
          </label>
          <input
            checked={true}
            className="form-control m-1"
            id="exampleInputEmail1"
            name="name"
            onChange={[Function]}
            type="text"
            value="element summary"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
